backend/portal/api/curation/v1/curation/collections/collection_id/datasets/dataset_id/actions.py

```diff
@@ -1,41 +1,161 @@
+    if collection_version is None:
         raise NotFoundHTTPException("Collection not found!")
-    dataset = get_dataset_else_error(db_session, dataset_id, collection_id)
-    response_body = reshape_dataset_for_curation_api(dataset.to_dict_keep(EntityColumns.columns_for_dataset))
+
+    version = get_infered_dataset_version(dataset_id)
+    if version is None:
+        raise NotFoundHTTPException("Dataset not found")
+
+    response_body = _reshape_dataset_for_curation_api(version)
     return make_response(jsonify(response_body), 200)
 
 
-@dbconnect
 def delete(token_info: dict, collection_id: str, dataset_id: str = None):
-    db_session = g.db_session
-    dataset = get_dataset_else_error(db_session, dataset_id, collection_id, include_tombstones=True)
-    delete_dataset_common(db_session, dataset, token_info)
-    return "", 202
+
+    business_logic = get_business_logic()
+    user_info = UserInfo(token_info)
+
+    # TODO: deduplicate from ApiCommon. We need to settle the class/module level debate before can do that
+    dataset_version = business_logic.get_dataset_version(DatasetVersionId(dataset_id))
+    if dataset_version is None:
+        raise ForbiddenHTTPException(f"Dataset {dataset_id} does not exist")
+
+    collection_version = business_logic.get_collection_version(CollectionVersionId(collection_id))
+    # If the collection does not exist, it means that the dataset is orphaned and therefore we cannot
+    # determine the owner. This should not be a problem - we won't need its state at that stage.
+    if collection_version is None:
+        raise ForbiddenHTTPException(f"Dataset {dataset_id} unlinked")
+    if not user_info.is_user_owner_or_allowed(collection_version.owner):
+        raise ForbiddenHTTPException("Unauthorized")
+    # End of duplicate block
+
+    # TODO: deduplicate from ApiCommon. We need to settle the class/module level debate before can do that
+    if dataset_version.version_id not in [v.version_id for v in collection_version.datasets]:
+        raise ForbiddenHTTPException(f"Dataset {dataset_id} does not belong to a collection")
+
+    try:
+        business_logic.remove_dataset_version(collection_version.version_id, DatasetVersionId(dataset_id))
+    except CollectionUpdateException:
+        raise MethodNotAllowedException(detail="Cannot delete a public Dataset")
+    return Response(status=202)
+    # End of duplicate block
 
 
 def put(collection_id: str, dataset_id: str, body: dict, token_info: dict):
-    upload_from_link(
-        collection_id,
-        token_info,
-        body.get("url", body.get("link")),
-        dataset_id,
-    )
-    return "", 202
+    # TODO: deduplicate from ApiCommon. We need to settle the class/module level debate before can do that
+    url = body.get("url", body.get("link"))
+    business_logic = get_business_logic()
+    dataset_version = business_logic.get_dataset_version(DatasetVersionId(dataset_id))
```

Comment:
Lin 130-136 is duplicated logic from a the delete handler. We should factor is out.